### PR TITLE
Add Mi100 support

### PR DESCRIPTION
**Pull request by gsitaram**
This commit contains changes to allow DBCSR to be built and run on AMD's MI100 GPUs. Building CP2K with DBCSR in it should also work.

### CODE REVIEW

**Comment by gsitaram** [comment]
I am having some problems compiling CP2K with this patch. Please do not merge it yet.

**Comment by hfp** [comment]
Approved CI tests.

**Comment by hfp** [comment]
I think you just need to run `black src/acc/libsmm_acc/tune/tune_setup.py` and update your PR. To run black, you can [install the precommit hooks](https://github.com/cp2k/dbcsr/wiki/Development) or get it stand-alone like `pip install black`.

File: src/acc/libsmm_acc/tune/tune_setup.py [hunk 1]
```diff
@@ -406,8 +406,9 @@ def gen_makefile(outdir, compiler, arch):
                 + " -w -o $@ $^ -lcuda -lnvrtc\n\n"
             )
         else:
+            rocm_path = os.getenv('ROCM_PATH', '/opt/rocm')
             output += (
-                "\thipcc -O3 -D__HIP -w -o $@ $^ /opt/rocm/hip/lib/libhiprtc.so\n\n"
+                "\thipcc -O3 -D__HIP -w -o $@ $^ {}/hip/lib/libamdhip64.so\n\n".format(rocm_path)
```
Comment (dev-zero) [inline src/acc/libsmm_acc/tune/tune_setup.py:411]:
```suggestion
                f"\thipcc -O3 -D__HIP -w -o $@ $^ {rocm_path}/hip/lib/libamdhip64.so\n\n"
```

**Comment by dev-zero** [comment]
CMake/make-part LGTM, thanks!

File: src/acc/libsmm_acc/tune/tune_setup.py (hunk 1 above)
Comment (gsitaram) [reply src/acc/libsmm_acc/tune/tune_setup.py]: Done, thank you for the pointer.

**Pull request merged**